worlds/landstalker/Hints.py

```diff
@@ -0,0 +1,127 @@
+from BaseClasses import MultiWorld
+from worlds.landstalker.data.hint_source import HINT_SOURCES_JSON
+
+
+def generate_blurry_location_hint(location, random):
+    cleaned_location_name = location.hint_text.lower().translate({ord(c): None for c in '(),:'})
+    cleaned_location_name.replace('-', ' ')
+    cleaned_location_name.replace('/', ' ')
+    cleaned_location_name.replace('.', ' ')
+    location_name_words = [w for w in cleaned_location_name.split(' ') if len(w) > 3]
+
+    random_word_1 = "mysterious"
+    random_word_2 = "place"
+    if len(location_name_words) > 0:
+        random_word_1 = random.choice(location_name_words)
+        location_name_words.remove(random_word_1)
+        if len(location_name_words) > 0:
+            random_word_2 = random.choice(location_name_words)
+    return [random_word_1, random_word_2]
+
+
+def generate_lithograph_hint(multiworld: MultiWorld, player: int):
+    hint_text = "It's barely readable:\n"
+
+    my_jewel_items = [item for item in multiworld.itempool if item.player == player and " Jewel" in item.name]
```

Comment:
Again, this would add a circular dependency between Hints and LandstalkerWorld.
I could pass a jewel_items array which would be built at that time, but it just moved stuff related to hints from Hints to __init__ with no significant performance boost from my tests.
I'd argue not to make that change, but remain open for suggestions on that matter :)

Reply:
Good point, I did it as a consequence.
Thanks